src/main/java/hello/controller/ApartmentUIController.java

```diff
@@ -0,0 +1,138 @@
+package hello.controller;
+
+import hello.exceptions.BadRequestException;
+import hello.exceptions.NotFoundException;
+import hello.model.Apartment2;
+import hello.service.ApartmentService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.ModelAttribute;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+
+
+
+
+
+@Controller
+public class ApartmentUIController {
+    private ApartmentService apartmentService;
+
+
+    @Autowired
+    public ApartmentUIController(ApartmentService apartmentService) {
+        this.apartmentService = apartmentService;
+    }
+
+
+    @GetMapping("/apartment")
+    public String getApartmentLandingPage(Model model) {
+        model.addAttribute("activeLink", "Apartment");
+        return "apartment";
+    }
+
+    @GetMapping("/findAnApartment")
+    public String showFindApartmentForm(Model model) {
+        Apartment2 apartment = new Apartment2();
+        model.addAttribute("apartment", apartment);
+        model.addAttribute("activeLink", "Apartment");
+        return "findAnApartment";
+    }
+
+    @PostMapping("/findApartmentResult")
+    public String showFindApartmentResult(@ModelAttribute("apartment") Apartment2 apartment, @RequestParam("id") String id, Model model) {
+        if (apartmentService == null) {
```

Comment:
Will this ever happen? If spring handles the injection of the service, you'd hope this would never evaluate to true!